sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/javaCode.scala

```diff
@@ -256,6 +283,22 @@ object EmptyBlock extends Block with Serializable {
   override def + (other: Block): Block = other
 }
 
+/**
+ * A block inlines all types of input arguments into a string without
+ * tracking any reference of `JavaCode` instances.
+ */
+case class InlineBlock(block: String) extends Block {
+  override val code: String = block
+  override val exprValues: Set[ExprValue] = Set.empty
+
+  override def + (other: Block): Block = other match {
+    case c: CodeBlock => Blocks(Seq(this, c))
+    case i: InlineBlock => InlineBlock(block + i.block)
+    case b: Blocks => Blocks(Seq(this) ++ b.blocks)
```

Comment:
nit: `this +: b.blocks`?

Reply:
After discussed with @cloud-fan, I think this abstraction `Blocks` is not necessary. It can be replaced with `CodeBlock`. I will submit a PR to do that.

Reply:
shall we do that PR first? I feel it's easier to review this PR after we clean up the `Block` framework.

Reply:
Ok. I will do that PR first. Will ping you on the PR when it's ready.